src/components/director-view/director-view.jsx

```diff
@@ -1,33 +1,80 @@
-import React from "react";
-import { useSelector } from "react-redux";
-import { useParams } from "react-router";
+import React, { useState, useEffect } from "react";
+import { Container, Row, Col } from "react-bootstrap";
+import { useParams, useNavigate } from "react-router-dom";
+import PropTypes from "prop-types";
 
+export const DirectorView = ({ token }) => {
+  const { directorName } = useParams();
+  const navigate = useNavigate();
+  const [director, setDirector] = useState(null);
 
-export const DirectorView = () => {
-  const params = useParams();
-  const directors = useSelector((state) => state.directors.list);
-  const director = directors.find((director) => director.name === params.name);
-
+  useEffect(() => {
+    if (!token) return;
+    (async () => {
+      try {
+        const response = await fetch(
+          `https://movieapi-aeueoes-projects.vercel.app/directors/${directorName}`,
```

Comment:
You can always store, the basic URL in the .environments file and access it from there, read more about environment variables in React

check the package `dotenv`